example-todomvc/src/main/scala/mhtml/todo/MhtmlTodo.scala

```diff
@@ -0,0 +1,342 @@
+//    Rx(load()) |@| header.model |@| todoListDataChanges
+//  ) map {
+//    case (stored: Seq[Todo], newTodoMaybe: Option[Todo], changes: List[TodoListItemData]) =>
+//      val removals: List[Todo] = changes.flatMap(change => change.removal)
+//      val updates: List[TodoUpdate] = changes.flatMap(change => change.update)
+//
+//      (stored ++ newTodoMaybe).filter(todo => !removals .contains(todo))
+//      .map{todo => updates.find{update => update.oldTodo == todo} match {
+//        case Some(foundUpdate) => foundUpdate.newTodo
+//        case None => todo
+//      }}
+//  }
+
+  case class ModelSources(
+    headerModel: Option[Todo],
+    changes:  List[TodoListItemData]
+  )
+
+  def updateState(currentTodos: Seq[Todo], sources: ModelSources): Seq[Todo] = {
+    val removals: List[Todo] = sources.changes.flatMap(change => change.removal)
+    val updates: List[TodoUpdate] = sources.changes.flatMap(change => change.update)
+
+    (currentTodos ++ sources.headerModel).filter(todo => !removals .contains(todo))
+      .map{todo => updates.find{update => update.oldTodo == todo} match {
+        case Some(foundUpdate) => foundUpdate.newTodo
+        case None => todo
+      }}
+  }
+
+  val sourcesWrapped: Rx[ModelSources] = header.model |@| todoListDataChanges map {
+    case(newTodoMaybe: Option[Todo], changes: List[TodoListItemData]) =>
+      ModelSources(newTodoMaybe, changes)
+    case whatsit  =>
+      println("mismatch: " + whatsit.getClass.getName)
+      throw new UndefinedBehaviorError("Expected (Option[Todo], List[TodoListItemData])")
+  }
+
+  lazy val allTodos: Rx[Seq[Todo]] = sourcesWrapped.flatMap(mSources =>
+    allTodos.foldp(load()){(last, next) => updateState(last ++ next, mSources)}
+  )
+
+
+  val autoSave: Unit = allTodos.map(save)
+
+//  }
+
+  // We're not strict about having all updates happen here, only the ones that
+  // contain some non-trivial logic.
+  //object Update {
+//  def newTodo(title: String): Unit =
+//    allTodos.update(todos => Todo(title, completed = false) +: todos)
+//  def removeTodo(todo: Todo): Unit =
+//    allTodos.update(_.filterNot(_ eq todo))
+//  def updateTodo(toUpdate: Todo, newTodo: Todo) =
+//    allTodos.update(todos => todos.updated(todos.indexOf(toUpdate), newTodo))
+  //}
+
+  //object View {
+
+  // helper to render <input checked={null}> as <input>
+  def conditionalAttribute(cond: Boolean) = if (cond) "true" else null
+
+
+
+  def focusInput() = dom.document.getElementById("editInput") match {
+    case t: HTMLInputElement => t.focus()
+    case _ =>
+  }
+
+
+  val count = active.items.map { items =>
+    <span class="todo-count">
+      <strong>{ items.length }</strong>
+      {if (items.length === 1) "item" else "items"} left
+    </span>
+  }
+
+  def todoListsFooter(todoList: TodoList) = {
+    val css = currentTodoList.map(x => if (x == todoList) "selected" else "")
+    <li>
+      <a href={ todoList.hash } class={css}>{ todoList.text }</a>
```

Comment:
Inconsistency with spaces around `{}`, I personally like it with the spaces but whatever, as long as it's consistent ;)